WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainCoordinator.swift

```diff
@@ -78,12 +78,14 @@ class RegisterDomainCoordinator {
                                shouldPush: false)
     }
 
-    func handleNoSiteChoice(on viewController: UIViewController) {
+    func handleNoSiteChoice(on viewController: RegisterDomainSuggestionsViewController) {
```

Comment:
[Suggestion]
I suggest passing the view model instead of specifying the view controller subclass, to maintain some level of abstraction.

This function can look like this:
```swift
func handleNoSiteChoice(on viewController: UIViewController,
                        choicesViewModel: DomainPurchaseChoicesViewModel?) {
    createCart(
        onSuccess: { [weak self] in
            self?.presentWebViewForNoSite(on: viewController)
            choicesViewModel?.isGetDomainLoading = false
        }) {
            viewController.displayActionableNotice(title: TextContent.errorTitle, actionTitle: TextContent.errorDismiss)
            choicesViewModel?.isGetDomainLoading = false
        }
}
```

And `pushPurchaseDomainChoiceScreen` can look like this:
```swift
private func pushPurchaseDomainChoiceScreen() {
    @ObservedObject var choicesViewModel = DomainPurchaseChoicesViewModel()
    let view = DomainPurchaseChoicesView(viewModel: choicesViewModel) { [weak self] in
        guard let self else { return }
        choicesViewModel.isGetDomainLoading = true
        self.coordinator?.handleNoSiteChoice(on: self, choicesViewModel: choicesViewModel)
    } chooseSiteAction: { [weak self] in
        guard let self else { return }
        self.coordinator?.handleExistingSiteChoice(on: self)
    }
    let hostingController = UIHostingController(rootView: view)
    hostingController.title = TextContent.domainChoiceTitle
    self.navigationController?.pushViewController(hostingController, animated: true)
}
```

Reusing handleNoSiteChoice on a different view controller will be as easy as passing `choicesViewModel` as nil, or changing this to a protocol and passing a different component.